### ISSUE TITLE
TargetSelector.hs: Irrefutable pattern failed for pattern t' : _

### ISSUE THREAD

**Issue by lambdageek** [haskell/cabal#4718]
```
$ cabal --version
cabal-install version 2.0.0.0
compiled using version 2.0.0.2 of the Cabal library 
```

```
$ cabal -v new-run
CallStack (from HasCallStack):
  die', called at ./Distribution/Client/CmdRun.hs:227:5 in main:Distribution.Client.CmdRun
cabal: Distribution/Client/TargetSelector.hs:(355,7)-(356,55): Irrefutable pattern failed for pattern t' : _
```

(The project in question is lambdageek/unbound-generics@f75cfdc62 after a successful `cabal new-build --enable-tests`)

**Comment by 23Skidoo** [comment]
IIRC this is fixed in `master`. 

/cc @fgaz

**Comment by hvr** [comment]
...and moreover, afaik, `new-run` is not fully implemented for 2.0 anyway...

**Comment by lambdageek** [comment]
Yea this was part of my occasional "can I use `new-foo` for my projects yet" exploration. 😄

**Comment by fgaz** [comment]
Yup, already fixed with #4583.
